kotlin/android/app/src/main/java/dev/firezone/android/features/auth/ui/AuthViewModel.kt

```diff
@@ -44,7 +50,7 @@ internal class AuthViewModel
                             ViewAction.LaunchAuthFlow(
                                 url =
                                     "${config.authBaseUrl}" +
-                                        "?client_csrf_token=$csrfToken&client_platform=android",
+                                        "?state=$state&nonce=$nonce&as=client",
```

Comment:
The reason it's okay to do string interpolation here is that we've just generated state and nonce 10 lines ago, so there can't be anything malicious in them, right?

Reply:
Correct, they're set as a constant (using `val =`) and so can't be changed.